### ISSUE TITLE
MLV-App QT uses all available memory on OSX

### ISSUE THREAD

**Issue by iFloris** [ilia3101/MLV-App#3]
Hello, yesterday I spent the day filming more raw video in 3k / 3.5k on my 5d3. In total I shot 64 short video's with a duration of between 20 and 180 seconds. I edited each of these MLV's in MLV-App, carefully avoiding the pink highlights issue by increasing the light strength.
I then set all MLV's to export overnight. This morning, when I went to see how far the process was, my Mac Pro was frozen. MLV-App had used all my systems' memory. 

[See this screenshot from iStat menus](https://www.dropbox.com/s/aswfwll0wx14i8w/Schermafdruk%202017-09-03%2010.50.45.jpg?dl=0)

As you can see, MLV-App tries to use a bit over 60 gigabyte memory. My Mac Pro has 32 gigabyte of physical memory, so Mac osx allocated as much virtual and compressed memory as it could. 

It probably is not meant to use that much memory, hence this report.

TL;DR
Set MLV-App QT to export 64 3k raw MLV's to mov on osx.
MLV-APP used all available memory, causing Mac osx to freeze.

**Comment by ilia3101** [comment]
I will have a look at what part of the code may be causing this inside the mlv libraries.

@masc4ii Could you check what Qt GUI isdoing while exporting? Is it enabling cache on every clip and forgetting to disable maybe? Basically we need to figure out which function call causes this disaster and if it can be prevented easily.

I can't imagine how such a huge memory leak could happen 😢 

And I apologise for the inconvenience of freezing your computer and the waste of time.

**Comment by iFloris** [comment]
@ilia3101 Apologies not needed.
As I wrote in #2 , ❤️ what you are doing with the app.
A freeze now and then is a small price to pay if that in some small way helps MLV-App become as great as you intend it to be! 🚀 
Also, it does say pre-alpha on the tin 😄

**Comment by ilia3101** [comment]
@iFloris Thanks 👍

**Comment by masc4ii** [comment]
Okaaaay. Hm... first we have to fix the bug opening lossless files - for me that does not work atm. Then we should have a look what happens, when exporting these files. Exporting normal mlv files causes no problem (I am doing that each evening the last week) - maximum RAM by MLV App is 150MB on my MacBook.

**Comment by ilia3101** [comment]
It opens lossless files from a camera completely fine, just not mlv_dump ones for some reason, I tested on the big file from iFloris and it was fine, but the small shortened ones didn't work.
And I just had another idea what the memory problem could be: possibly caused by using the lj92 library wrong?
@bouncyball-git do you have any idea why MLV-App won't open mlv_dump compressed mlvs? Despite the fact that we have switched back to the original lj92 lib

**Comment by ilia3101** [comment]
Ok, from a quick test, memory is being leaked when using the decompression library (no offence to the lj92 library, it's definitely my fault)

**TEST 1:**
- Opened iFloris's 500mb compressed clip
- let it cache
- then exported
- closed it
- 880mb of ram was still used!!! (now multiply this by 64 clips, and it becomes clear :rofl:)

**TEST 2:**
- Opened one of my own long clips
- Let it cache
- Exported
- closed it
- 135mb of ram still used

**Comment by masc4ii** [comment]
Can confirm your first result:
Your Test1 ... but I have then 1.1GB
Test2: on export or after, I am at 80-150MB

**Comment by ilia3101** [comment]
Will try out a few things and see if it can be prevented.

**Comment by ilia3101** [comment]
Ok it turns out I'm stupid, another case of me not freeing memory.
If an MLV clip is compressed, the memory used for reading RAW data is *never* freed, so it throws away 2-6MB reading every frame from file... so in the Qt app where caching is disabled(?) and it always reads frames from files, it is especially a worse problem.

**Comment by ilia3101** [comment]
Repeated test 1, only 110mb left afterwards, pretty certain it's fixed.

**Comment by iFloris** [comment]
Cool! 👏  I cannot wait to try it out later! I can compile it from source, you guys made it pretty simple 😄
